pkg/kube/inject/webhook.go

```diff
@@ -392,6 +419,88 @@ func injectPod(req InjectionParameters) ([]byte, error) {
 	return patch, nil
 }
 
+// OverrideAnnotation is used to store the overrides for injected containers
+// TODO move this to api repo
+const OverrideAnnotation = "proxy.istio.io/overrides"
+
+// reapplyOverwrittenContainers enables users to provide container level overrides for settings in the injection template
+// * originalPod: the pod before injection. If needed, we will apply some configurations from this pod on top of the final pod
+// * templatePod: the rendered injection template. This is needed only to see what containers we injected
+// * finalPod: the current result of injection, roughly equivalent to the merging of originalPod and templatePod
+// There are essentially three cases we cover here:
+// 1. There is no overlap in containers in original and template pod. We will do nothing.
+// 2. There is an overlap (ie, both define istio-proxy), but that is because the pod is being re-injected.
+//    In this case we do nothing, since we want to apply the new settings
+// 3. There is an overlap. We will re-apply the original container.
```

Comment:
Can you give an example for 3?

Reply:
Yes, the added test cases in this PR are examples of this case (like `proxy-override.yaml`)

Reply:
Ok thanks ... could we just add a blurb here in the comment to explain what we mean by overlap in this case. IIUC the sort of overlap we're talking about is when a user customized the arguments to the istio-init or istio-proxy containers?

Reply:
Yeah exactly. Let me expand this a bit

Reply:
added